.github/workflows/main.yml

```diff
@@ -0,0 +1,53 @@
+name: Pull Request
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on pull request events for the develop and master branches
+  pull_request:
+    branches: [ develop, master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "test"
+  test:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: Build and Test project
+        run: |
+          echo Build and Test project
+          ./mvnw clean verify
+
+      - name: Coverage Information
+        id: coverage_info
+        run: |
+          per_package_info=$(awk -F "," '/three-hundred-sixty/ { missed[$2] += $4 } { covered[$2] += $5  } END { for(x in missed)printf ("%s %s%%\n", x, (covered[x]/(covered[x]+missed[x]))*100) }' target/site/jacoco/jacoco.csv)
+          per_package_info="${per_package_info//'%'/'%25'}"
+          per_package_info="${per_package_info//$'\n'/'%0A'}"
+          per_package_info="${per_package_info//$'\r'/'%0D'}"
+
+          overall_info=$(awk -F "," '/three-hundred-sixty/ { missed[$1] += $4 } { covered[$1] += $5  } END { for(x in missed)printf ("Overall coverage: %s%%\n", (covered[x]/(covered[x]+missed[x]))*100) }' target/site/jacoco/jacoco.csv)
```

Comment:
The `awk` command runs on a file that looks like this:

```
GROUP,PACKAGE,CLASS,INSTRUCTION_MISSED,INSTRUCTION_COVERED,BRANCH_MISSED,BRANCH_COVERED,LINE_MISSED,LINE_COVERED,COMPLEXITY_MISSED,COMPLEXITY_COVERED,METHOD_MISSED,METHOD_COVERED
three-hundred-sixty,dev.drugowick.threehundredsixty.validation,NewPasswordFieldsMustMatchValidator,10,47,1,1,4,11,1,3,0,3
three-hundred-sixty,dev.drugowick.threehundredsixty.validation,ImprovementNotBlankIfEvaluationIsNegativeValidator,87,0,8,0,20,0,7,0,3,0
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.entity,FeedbackRelationship,0,56,0,0,0,8,0,2,0,2
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.entity,Employee,0,6,0,0,0,1,0,1,0,1
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.entity,FeedbackState,0,56,0,0,0,8,0,2,0,2
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.entity,Question,0,18,0,0,0,7,0,1,0,1
three-hundred-sixty,dev.drugowick.threehundredsixty.config,BootstrapData,7,403,3,1,0,115,2,4,0,4
three-hundred-sixty,dev.drugowick.threehundredsixty.config,WebSecurityConfig,0,65,0,0,0,14,0,4,0,4
three-hundred-sixty,dev.drugowick.threehundredsixty.config,ValidationConfig,0,12,0,0,0,4,0,2,0,2
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.util,FeedbackInput,3,0,0,0,1,0,1,0,1,0
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.entity.security,MyUserDetails,9,36,0,0,4,11,4,4,4,4
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.admin,EmployeeAdminController,0,181,0,6,0,46,0,15,0,12
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.admin,FeedbackReportBetaController,68,15,2,0,17,6,4,1,3,1
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.admin,QuestionAdminController,72,6,4,0,19,3,9,1,7,1
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.admin,FeedbackAdminController,168,15,2,0,46,6,8,1,7,1
three-hundred-sixty,dev.drugowick.threehundredsixty.controller.profile,PasswordChangeController,0,64,1,5,0,17,1,6,0,4
three-hundred-sixty,dev.drugowick.threehundredsixty,ThreeHundredSixtyApplication,0,8,0,0,0,3,0,2,0,2
three-hundred-sixty,dev.drugowick.threehundredsixty.dto,AnswerDto,3,0,0,0,1,0,1,0,1,0
three-hundred-sixty,dev.drugowick.threehundredsixty.service,MyUserDetailsService,6,22,0,0,0,6,1,2,1,2
three-hundred-sixty,dev.drugowick.threehundredsixty.service,ReportsServiceImpl,194,9,10,0,37,4,7,1,2,1
three-hundred-sixty,dev.drugowick.threehundredsixty.service,FeedbackServiceImpl,8,52,0,0,3,18,0,3,0,3
three-hundred-sixty,dev.drugowick.threehundredsixty.domain.dto,QuestionPositionScore,25,0,0,0,8,0,1,0,1,0
three-hundred-sixty,dev.drugowick.threehundredsixty.controller,FeedbackController,0,51,0,2,0,13,0,4,0,3
three-hundred-sixty,dev.drugowick.threehundredsixty.controller,HomePageController,23,6,0,0,5,3,1,1,1,1
three-hundred-sixty,dev.drugowick.threehundredsixty.controller,BaseController,3,55,1,7,1,13,1,8,0,5
three-hundred-sixty,dev.drugowick.threehundredsixty.controller,ErrorController,1,24,1,1,0,6,1,3,0,3
three-hundred-sixty,dev.drugowick.threehundredsixty.controller,QuestionController,171,9,8,0,41,4,11,1,7,1
```

That's what it does:
 
- defines comma as field separator;
- applies to lines starting with `three-hundred-sixty` only;
- sets an array called `missed` keyed by the first field (group `three-hundred-sixty`), which sums all the missed instructions for this group;
- sets an array called `covered` keyed by the first field (group `three-hundred-sixty`), which sums all the covered instructions for this group;
- prints each group with the percent of lines covered.

**NOTE**: the other `awk` command is about the same, but it groups on the package instead of the group, to print detailed coverage information later.

Reply:
Thanks for the quite detailed explanation here, @brunodrugowick !
I had no clue what was the file format until see this snippet, otherwise, your `awk` would look like a magic spell with these magic numbers (the groups id, eg `$2`, `$4` and so on)

Reply:
`awk` always looks like magic... even the ones I make after a few weeks seem like magic to me. 😆